{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "71688404-4b26-4408-956e-b20cd3b02993",
   "metadata": {
    "editable": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('49418.0', '1.0', '9782378.0', '1:9782378', 'G', 'Gly', 'A', 'Ser', 'missense', '1838.0', 'Unknown significance', 'Pending', 'PIK3CD', '+', '18.0', 'NM_005026.3', 'NP_005017.3', '2311.0', '1.0', '771.0', 'g.9782378G>A', 'NM_005026.3', 'c.2311G>A', 'NP_005017.3', 'p.Gly771Ser', 'Heme-STAMP', '45294.729166666664', '2024/01/03: gnomAD: South Asian maf=0.000360, Other maf=0.000165, Latino maf=0.000029, Total maf=0.000056. ClinVar: Conflicting interpretations of pathogenicity, 1 star, 2 submissions. Not in JaxCKB. No functional data in published literature. OncoKB: Unknown.', None), ('49417.0', '13.0', '3.2913512E7', '13:32913512', 'A', 'Ser', 'G', 'Gly', 'missense', '1044.0', 'Unknown significance', 'Pending', 'BRCA2', '+', '11.0', 'NM_000059.3', 'NP_000050.2', '5020.0', '1.0', '1674.0', 'g.32913512A>G', 'NM_000059.3', 'c.5020A>G', 'NP_000050.2', 'p.Ser1674Gly', 'Capp-Seq', '45294.28888888889', '2024/01/02: gnomAD: African maf=0.000458, Latino maf=0.000125, European maf=0.000000, Total maf=0.000056. ClinVar: Conflicting interpretations of pathogenicity, 1 star, 13 submissions. Not in JaxCKB. No functional data in published literature. OncoKB: Unknown.', None), ('49416.0', '13.0', '3.291267E7', '13:32912670', 'C', 'Ala', 'T', 'Val', 'missense', '1188.0', 'Unknown significance', 'Pending', 'BRCA2', '+', '11.0', 'NM_000059.3', 'NP_000050.2', '4178.0', '2.0', '1393.0', 'g.32912670C>T', 'NM_000059.3', 'c.4178C>T', 'NP_000050.2', 'p.Ala1393Val', 'Capp-Seq', '45294.28888888889', '2024/01/02: gnomAD: African maf=0.000064, European maf=0.000000, Finnish maf=0.000000, Total maf=0.000004. ClinVar: Conflicting interpretations of pathogenicity, 1 star, 9 submissions. Not in JaxCKB. No functional data in published literature. OncoKB: Unknown.', None), ('49415.0', '12.0', '6.9233344E7', '12:69233344', 'A', 'Pro', 'G', 'Pro', 'synonymous', '1403.0', 'Unknown', 'Pending', 'MDM2', '+', '11.0', 'NM_002392.5', 'NP_002383.2', '1209.0', '3.0', '403.0', 'g.69233344A>G', 'NM_002392.5', 'c.1209A>G', 'NP_002383.2', 'p.Pro403=', 'Capp-Seq', None, None, None), ('49414.0', '8.0', '6.7485719E7', '8:67485719', 'A', 'Asn', 'G', 'Ser', 'missense', '742.0', 'Unknown significance', 'Pending', 'MYBL1', '-', '11.0', 'NM_001080416.2', 'NP_001073885.1', '1493.0', '2.0', '498.0', 'g.67485719T>C', 'NM_001080416.2', 'c.1493A>G', 'NP_001073885.1', 'p.Asn498Ser', 'Capp-Seq', '45294.288194444445', '2024/01/02: gnomAD: African maf=0.000416, Latino maf=0.000060, European maf=0.000000, Total maf=0.000044. ClinVar: Uncertain significance, 1 star, 1 submission. Not in JaxCKB. No functional data in published literature. OncoKB: Unknown.', None), ('49413.0', '3.0', '1.78937422E8', '3:178937422', 'T', 'Cys', 'C', 'Arg', 'missense', '806.0', 'Likely Pathogenic', 'Pending', 'PIK3CA', '+', '12.0', 'NM_006218.2', 'NP_006209.2', '1810.0', '1.0', '604.0', 'g.178937422T>C', 'NM_006218.2', 'c.1810T>C', 'NP_006209.2', 'p.Cys604Arg', 'Capp-Seq', '45294.27847222222', 'The p.Cys604Arg (c.1810T>C, p.C604R) variant in the PIK3CA gene results in a substitution of an arginine for the wild-type cysteine at codon 604 and is located in the accessory domain of the protein. Expression of this mutation in Ba/F3 and MCF10A cell lines demonstrated that it is activating as measured by increased proliferation in the absence of growth factors compared to wildtype PIK3CA (PMID: 29533785). This variant has been recurrently identified in patients with colorectal adenocarcinoma (PMID: 27149842, PMID: 25344691). Tumor-associated PIK3CA mutations occur most commonly in the helical and kinase domains, and are associated with increased kinase activity and oncogenic transformation in vitro. They are found in approximately 30% of breast cancers (more commonly in estrogen/progesterone receptor positive and HER2 positive than triple negative tumors). PIK3CA mutations are also seen in some gynecologic malignancies, 10-30% of colorectal carcinomas, 10% of squamous non-small cell lung cancers, and fewer lung adenocarcinomas. In addition, activating mutations in the helical and kinase domains of PIK3CA are reported in lymphatic and venous malformations. mTOR inhibitors, such as sirolimus, are effective therapeutic options for patients with extensive vascular malformations (PMID 30373605, 29924669). References: PMID 22330809, 19671852, 22271473, 23936763, 18676830, 23612454, 17376864, 26655088, 21266528, 26492180.', None), ('49412.0', '3.0', '1.78916944E8', '3:178916944', 'A', 'Lys', 'G', 'Glu', 'missense', '855.0', 'Likely Pathogenic', 'Pending', 'PIK3CA', '+', '2.0', 'NM_006218.2', 'NP_006209.2', '331.0', '1.0', '111.0', 'g.178916944A>G', 'NM_006218.2', 'c.331A>G', 'NP_006209.2', 'p.Lys111Glu', 'Capp-Seq', '45294.28680555556', \"The p.Lys111Glu (c.331A>G, p.K111E) variant in the PIK3CA gene results in a substitution of a glutamic acid for the wild-type lysine at codon 111 and is located in the protein's linker region within the N-terminus. This mutation has been found to occur in endometrial and breast cancers (PMID: 18074223, 22949682, 21266528). Expression of this mutation in Ba/F3 and MCF10A cell lines demonstrated that it is activating as measured by increased downstream pathway activation and proliferation in the absence of growth factors compared to wildtype PIK3CA (PMID: 29533785, 18074223, 22949682, 21266528). Tumor-associated PIK3CA mutations occur most commonly in the helical and kinase domains, and are associated with increased kinase activity and oncogenic transformation in vitro. They are found in approximately 30% of breast cancers (more commonly in estrogen/progesterone receptor positive and HER2 positive than triple negative tumors). PIK3CA mutations are also seen in some gynecologic malignancies, 10-30% of colorectal carcinomas, 10% of squamous non-small cell lung cancers, and fewer lung adenocarcinomas. In addition, activating mutations in the helical and kinase domains of PIK3CA are reported in lymphatic and venous malformations. mTOR inhibitors, such as sirolimus, are effective therapeutic options for patients with extensive vascular malformations (PMID 30373605, 29924669). References: PMID 22330809, 19671852, 22271473, 23936763, 18676830, 23612454, 17376864, 26655088, 21266528, 26492180.\", None), ('49411.0', '17.0', '7579699.0', '17:7579699', 'G', 'N/A', 'T', 'N/A', 'splice-site', '182.0', 'Likely Pathogenic', 'Pending', 'TP53', '-', 'intron', 'NM_000546.5', 'NP_000537.3', 'N/A', 'N/A', 'N/A', 'g.7579699C>A', 'NM_000546.5', 'c.96+1G>T', 'N/A', 'N/A', 'Capp-Seq', '45294.20277777778', ' The c.96+1G>T (p.?) variant in the TP53 gene results in disruption of a canonical splice site. Experimental studies of the novel transcript are not available. However, this mutation may result in exon skipping, retention of intronic regions in the transcript, and/or a premature stop codon. TP53 is the most commonly mutated gene in human cancer and mutations are typically associated with poor prognosis. References: PMID 23263379, 22713868, 21045690, 29979965. ', None), ('49410.0', '12.0', '1.33245468E8', '12:133245468', 'G', 'Glu', 'A', 'Lys', 'missense', '1232.0', 'Unknown significance', 'Pending', 'POLE', '-', '17.0', 'NM_006231.2', 'NP_006222.2', '1852.0', '1.0', '618.0', 'g.133245468C>T', 'NM_006231.2', 'c.1852G>A', 'NP_006222.2', 'p.Glu618Lys', 'Capp-Seq', '45294.19305555556', '2024/01/02: Not in gnomAD. ClinVar: Uncertain significance, 2 stars, 4 submissions. Not in JaxCKB. No functional data in published literature. OncoKB: Unknown.', None), ('49409.0', '11.0', '1.25495711E8', '11:125495711', 'C', 'N/A', 'G', 'N/A', 'N/A', '728.0', 'Unknown', 'Pending', 'CHEK1', '+', \"5' UTR\", 'NM_001274.5', 'NP_001265.2', 'N/A', 'N/A', 'N/A', 'g.125495711C>G', 'NM_001114121.2', 'c.-217C>G', 'N/A', 'N/A', 'Capp-Seq', '45294.191666666666', '2024/01/02: Intronic. gnomAD: East Asian maf=0.000096, European maf=0.000000, Finnish maf=0.000000, Total maf=0.000008. Not in ClinVar. Not in JaxCKB. No functional data in published literature. OncoKB: Unknown.', None)]\n",
      "[(3,)]\n"
     ]
    }
   ],
   "source": [
    "import duckdb\n",
    "\n",
    "con = duckdb.connect()\n",
    "con.sql(\"CREATE TABLE new_tbl AS SELECT * FROM read_xlsx('../data/example_variant_data.xlsx', all_varchar = true);\")\n",
    "result = con.execute(\"SELECT * FROM new_tbl LIMIT 10;\").fetchall()\n",
    "no_unknown = con.execute(\"SELECT * FROM new_tbl WHERE Pathogenicity != 'Unknown' LIMIT 10\").fetchall()\n",
    "count_types = con.execute(\"SELECT count(DISTINCT Pathogenicity) FROM new_tbl;\").fetchall()\n",
    "print(result)\n",
    "print(count_types)\n",
    "con.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40143f9d-761d-4ce1-8dc7-ff4672061bfe",
   "metadata": {
    "editable": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": []
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
